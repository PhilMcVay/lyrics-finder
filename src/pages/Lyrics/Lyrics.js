import React from "react";

import { LyricsContainer, Line } from "./Lyrics.style";

const formatLyrics = (lyrics) => {
  const splitLyrics = lyrics.split("\n");

  return splitLyrics.map((line, index) => {
    if (!line.length) return <br />;
    return <Line key={`${line}-${index}`}>{line}</Line>;
  });
};

const Lyrics = ({ lyrics }) => (
  <LyricsContainer>{formatLyrics(lyrics)}</LyricsContainer>
);

export default Lyrics;
