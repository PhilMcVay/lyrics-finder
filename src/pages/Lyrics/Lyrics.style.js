import styled from "styled-components";

export const LyricsContainer = styled.section`
  overflow-y: scroll;
  height: calc(100% - 85px);
  padding-left: 8px;
`;

export const Line = styled.p`
  margin: 2px;
  font-size: 13px;
`;
