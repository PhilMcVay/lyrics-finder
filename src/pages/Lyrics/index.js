import React from "react";
import { useParams } from "react-router-dom";
import { useLyrics } from "../../hooks/useLyrics";

import Card from "../../shared/Card";
import { Container, ArtistName, BackLink } from "../../shared/styles";

import Lyrics from "./Lyrics";

const LyricsPage = () => {
  const { artistName, track } = useParams();
  const { lyrics, isLoading, error } = useLyrics(artistName, track);

  return (
    <Card>
      <Container data-testid="lyrics-page">
        <BackLink to="/">Back to Track List</BackLink>
        <ArtistName>
          {artistName} - {track}
        </ArtistName>
        {isLoading && <p>Loading Lyrics...</p>}
        {error && <p>No lyrics found for {track}</p>}
        {lyrics && <Lyrics lyrics={lyrics} />}
      </Container>
    </Card>
  );
};

export default LyricsPage;
