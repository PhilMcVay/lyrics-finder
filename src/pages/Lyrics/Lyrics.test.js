import React from "react";
import { render, screen } from "@testing-library/react";

import Lyrics from "./Lyrics";

const mockLyrics =
  "line 1\nline 2\nline 3\nline 4\n\nline 5\nline 6\nline 7\nline 8";

describe("<Lyrics>", () => {
  it("Should Render", () => {
    const { container } = render(<Lyrics lyrics={mockLyrics} />);

    expect(container).toBeInTheDocument();
  });

  it("Should format the lyrics correctly", () => {
    render(<Lyrics lyrics={mockLyrics} />);

    expect(screen.getByTestId("lyrics")).toMatchSnapshot();
  });
});
