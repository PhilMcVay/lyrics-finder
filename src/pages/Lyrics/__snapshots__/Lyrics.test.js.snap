// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Lyrics> Should format the lyrics correctly 1`] = `
<section
  class="sc-bdnylx eBbEUU"
  data-testid="lyrics"
>
  <p
    class="sc-gtssRu fLfXMm"
  >
    line 1
  </p>
  <p
    class="sc-gtssRu fLfXMm"
  >
    line 2
  </p>
  <p
    class="sc-gtssRu fLfXMm"
  >
    line 3
  </p>
  <p
    class="sc-gtssRu fLfXMm"
  >
    line 4
  </p>
  <br />
  <p
    class="sc-gtssRu fLfXMm"
  >
    line 5
  </p>
  <p
    class="sc-gtssRu fLfXMm"
  >
    line 6
  </p>
  <p
    class="sc-gtssRu fLfXMm"
  >
    line 7
  </p>
  <p
    class="sc-gtssRu fLfXMm"
  >
    line 8
  </p>
</section>
`;
